src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -527,8 +527,6 @@ SHAMapStoreImp::dbPaths()
             writableDbExists = true;
         else if (! state.archiveDb.compare (it->path().string()))
             archiveDbExists = true;
-        else if (! dbPrefix_.compare (it->path().stem().string()))
-            boost::filesystem::remove_all (it->path());
```

Comment:
This change can lead to the data directory being cluttered if a directories that are not active as either the writable or archive db are not cleaned up when the process starts. I'm not sure how you managed to get an extra directory in the data directory--under normal circumstances, there should only be the 2 directories with a small window of time in which a 3rd directory is brought into existence and the archive db is deleted. If the process crashes during that window of time, it is possible for there to be 3 directories. That is why the code you are trying to remove exists.

I don't know what you intend by backing up one of these directories. However, if you want to do something like back up the archive database while rippled is running, then you can activate online_delete manually by enabling the advisory_delete option in the config, and then only executing can_delete when you are ready to allow data to be deleted. The behavior of online_delete with these options is explained in detail here: https://xrpl.org/online-deletion.html

I believe that the change that you suggest will cause unintended consequences, and also what you are trying to do can be better accomplished by using the mechanisms described in that document. Thank you very much for your submission. I hope this explanation helps you to accomplish your objectives and also helps you to learn about how we implemented this feature.

Reply:
@mtrippled Instead of remove the codes as i said; there could be multiple conditions in comparing  `dbPrefix_` for do not delete the files if database path changed? Its actually working as expected but i think it shouldn't work when database path changed because we need to re-download everything and sync again and i dont know how users can make backups after rippled message if these folders are gone already.